src/commands/dev/lib/yarn-installer.ts

```diff
@@ -0,0 +1,71 @@
+import execa from 'execa';
+import { tmpdir } from 'os';
+import {
+  pathExists,
+  mkdirp,
+  createWriteStream,
+  statSync,
+  chmodSync
+} from 'fs-extra';
+import pipe from 'promisepipe';
+import { join } from 'path';
+import fetch from 'node-fetch';
+import tar from 'tar-fs';
+import { setScriptGlobals } from '@now/build-utils';
+import { Output } from '../../../util/output/create-output';
+
+async function isFoundInPath(): Promise<boolean> {
+  const [npm, yarn] = await Promise.all([
+    execa('npm', ['--version']),
+    execa('yarn', ['--version'])
+  ]);
+  return typeof npm === 'string' || typeof yarn === 'string';
+}
+
+async function installYarn(output: Output): Promise<string> {
+  // Loosely based on https://yarnpkg.com/install.sh
+  const dirName = join(tmpdir(), 'co.zeit.now', 'dev', 'yarn');
+  // const full = join(dirName, 'yarn-latest.tar.gz');
+  const yarnBin = join(dirName, 'bin', 'yarn');
+
+  if (await pathExists(yarnBin)) {
```

Comment:
We need to do a shasum check here instead of just checking if it exists. Consider a half-downloaded file, where `now dev` was Ctrl+C while downloading it. We need to recover from that gracefully.

Reply:
@TooTallNate Great idea! I fixed it!